KeenClient/KIODBStore.m

```diff
@@ -75,17 +81,11 @@ - (BOOL)openDB {
         keen_io_sqlite3_shutdown();
         keen_io_sqlite3_config(SQLITE_CONFIG_MULTITHREAD);
         keen_io_sqlite3_initialize();
-        
-        int openDBResult = keen_io_sqlite3_open([my_sqlfile UTF8String], &keen_dbname);
+	
+        int openDBResult = keen_io_sqlite3_open([dbFile UTF8String], &keen_dbname);
         if (openDBResult != SQLITE_OK) {
             if(openDBResult == SQLITE_CORRUPT) {
-                // delete corrupt database file
-                NSError *error = nil;
-                NSLog(@"%@", my_sqlfile);
-                [[NSFileManager defaultManager] removeItemAtPath:my_sqlfile error:&error];
-                
-                // create new database file
-                [self openDB];
```

Comment:
Thanks for getting rid of this recursive call, since as we talked about, it mostly would have deadlocked.